src/backend/distributed/deparser/deparse_role_stmts.c

```diff
@@ -380,12 +401,35 @@ AppendGrantRoleStmt(StringInfo buf, GrantRoleStmt *stmt)
 	{
 #if PG_VERSION_NUM >= PG_VERSION_16
 		DefElem *opt = NULL;
+		 int opt_count = 0;
 		foreach_ptr(opt, stmt->opt)
 		{
-			if (strcmp(opt->defname, "admin") == 0)
-			{
-				appendStringInfo(buf, " WITH ADMIN OPTION");
+			switch (opt->defname)
+        {
+            case "admin":
+                appendStringInfo(buf, " WITH ADMIN OPTION");
+                opt_count++;
+                break;
+
+			case "inherit":
+				if (opt_count > 0)
+				{
+					appendStringInfo(buf, ", ");
+				}
+				appendStringInfo(buf, "INHERIT OPTION ");
```

Comment:
Here you need to check the value of opt before deciding to use INHERIT OPTION. We may need to use INHERIT FALSE if opt's value is actually false. You can also check #7191 for reference on how to obtain opt's value.

Reply:
Is this just for INHERIT or for SET as well?